### PR TITLE
brocade fabric os enhancement 

### PR DESCRIPTION
**Pull request by crcro**
- changed the fabric os switch from network to storage (they are fiber-channels switches)
- added the known platforms to polling detection

### CODE REVIEW

File: includes/definitions.inc.php [hunk 1]
```diff
@@ -888,8 +888,8 @@
 $config['os'][$os]['over'][2]['text']  = 'Memory Usage';
 
 $os = 'fabos';
-$config['os'][$os]['text']             = 'Brocade FabricOS';
-$config['os'][$os]['type']             = 'network';
+$config['os'][$os]['text']             = 'FabricOS';
+$config['os'][$os]['type']             = 'storage';
```
Comment (laf) [inline includes/definitions.inc.php]: Why storage?

File: includes/polling/os/fabos.inc.php [hunk 2]
```diff
@@ -1,4 +1,199 @@
...
-$hardware = trim(snmp_get($device, 'ENTITY-MIB::entPhysicalDescr.1', '-Ovq'), '"');
+$gethardware = trim(snmp_get($device, 'SNMPv2-SMI::mib-2.75.1.1.4.1.3.1', '-Ovq'), '"');
+$revboard = str_replace("SNMPv2-SMI::enterprises.1588.2.1.1.", "", $gethardware);
+if(strpos($revboard, ".") !== false) {
+	$getid = strstr(str_replace($revboard, "", $gethardware), ".", true);
+} else {
+	$getid = $revboard;
+}
+
+switch($getid)
```
Comment (laf) [inline includes/polling/os/fabos.inc.php]: You should move this logic to: includes/rewrites.php as per other vendors hardware rewriting.

File: includes/definitions.inc.php (hunk 1 above)
Comment (crcro) [inline includes/definitions.inc.php]: because they are FC switches and usually they get under the administration of storage "department" not networking

Comment (laf) [inline includes/definitions.inc.php]: Anyone else got a comment on that - seems like it belongs in networking to me still as it's a switch.

@librenms/reviewers

Comment (f0o) [inline includes/definitions.inc.php]: Being an FC Switch I would also say that it's more Storage than Network

Comment (laf) [inline includes/definitions.inc.php]: I'm not overly fussed by this change but if it was up to me then I wouldn't change it to storage as the device provides no storage facilities.

Comment (crcro) [inline includes/definitions.inc.php]: actually it provides storage for FC connected hosts, and nothing else. you cannot do ethernet traffic on these switch for network communications. if it does not provide storage, it does not provide anything.

Comment (laf) [inline includes/definitions.inc.php]: But it doesn't provide storage itself does it, it provides connectivity to a storage platform.

Comment (crcro) [inline includes/definitions.inc.php]: still, being a "switch" it's not making it networking. maybe the categories are too loose in definition and this is way we are having this discussion :) in my opinion FC switches (that cannot provide ethernet connectivity) are storage infrastructure, not networking.

Comment (laf) [inline includes/definitions.inc.php]: Indeed, discussions are good.

Me personally, my mind is set and the example I gave in irc is that Cisco MDS fibre channel switches - the same category you are talking about here - are based on NX-OS which is cross platform so you can't then have some devices labelled as storage and some networking. It just doesn't make sense.

I won't be merging this as is, other @librenms/reviewers are more than welcome to feed into the discussion and merge if they feel necessary.

**Comment by crcro** [comment]
new PR #3988 without device category change

**Pull request closed without merging**